internal/test/e2e/run.go

```diff
@@ -150,18 +163,26 @@ type instanceRunConf struct {
 	testReportFolder, branchName                                              string
 	ipPool                                                                    networkutils.IPPool
 	hardware                                                                  []*api.Hardware
+	hardwareCount                                                             int
 	bundlesOverride                                                           bool
 	testRunnerType                                                            TestRunnerType
 	testRunnerConfig                                                          TestInfraConfig
 	cleanupVms                                                                bool
 	logger                                                                    logr.Logger
 }
 
-func RunTests(conf instanceRunConf) (testInstanceID string, testCommandResult *testCommandResult, err error) {
+//nolint:gocyclo, revive // RunTests responsible launching test runner to run tests is complex.
+func RunTests(conf instanceRunConf, hardwareQueue *hardwareQueue) (testInstanceID string, testCommandResult *testCommandResult, err error) {
 	testRunner, err := newTestRunner(conf.testRunnerType, conf.testRunnerConfig)
 	if err != nil {
 		return "", nil, err
 	}
+	if conf.hardwareCount > 0 {
```

Comment:
I guess this is kinda checking if its a tinkerbell test. But just by looking at this If its hard to know that. Can we have a method that does isTinkerbellTest() and that can internally check this?